intra_process_demo/README.md

```diff
@@ -0,0 +1,165 @@
+
+```bash
+# Open new terminal
+ros2 run intra_process_demo image_pipeline_with_two_image_views
+```
+
+![](img/image_pipeline_with_two_image_views_rqtgraph.png)
+
+## **Verify**
+
+### 1. Two Node Pipeline
+
+When executed correctly, strings should be printed in the terminal similar to what is shown below:
+
+```bash
+Published message with value: 0, and address: 0x55B68BCC6F20
+ Received message with value: 0, and address: 0x55B68BCC6F20
+Published message with value: 1, and address: 0x55B68BCC6F20
+ Received message with value: 1, and address: 0x55B68BCC6F20
+Published message with value: 2, and address: 0x55B68BCC6F20
+ Received message with value: 2, and address: 0x55B68BCC6F20
+Published message with value: 3, and address: 0x55B68BCC6F20
+ Received message with value: 3, and address: 0x55B68BCC6F20
+```
+
+### 2. Cyclic Pipeline
+
+When executed correctly, strings should be printed in the terminal similar to what is shown below:
+
+```bash
+Published first message with value:  42, and address: 0x555E4F029480
+Received message with value:         42, and address: 0x555E4F029480
+  sleeping for 1 second...
+  done.
+Incrementing and sending with value: 43, and address: 0x555E4F029480
+Received message with value:         43, and address: 0x555E4F029480
+  sleeping for 1 second...
+  done.
+Incrementing and sending with value: 44, and address: 0x555E4F029480
+Received message with value:         44, and address: 0x555E4F029480
+  sleeping for 1 second...
+  done.
+Incrementing and sending with value: 45, and address: 0x555E4F029480
+Received message with value:         45, and address: 0x555E4F029480
+  sleeping for 1 second...
+
+```
+
+### 3. Image Pipeline All In One
+
+When executed correctly, an OpenCV GUI window should appear displaying something similar to what is shown below:
+
+![](img/image_pipeline_all_in_one.png)
+
+>Take note how the **process_id** and **Message Pointer Address** are **all the same**, proving that all 3 nodes are in the same process.
+
+### 4. Image Pipleline All Separately
+
+When executed correctly, an OpenCV GUI window should appear displaying something similar to what is shown below:
+
+![](img/image_pipeline_all_separately.png)
+
+> Notice how all the **process_id** and **Message Pointer Address** are now **all different**, showing that all nodes are using different processes.
+
+### 5. Image Pipeline With Two Image Views
+
+When executed correctly, 2 OpenCV GUI window should appear displaying similar to what is show below:
+
+![](img/image_pipeline_with_two_image_views.png)
+
+> For more details on this implementations, please refer to the references below.
+
+## **References**
+
+1. [Intra Process Communication tutorial](https://daobook.github.io/ros2-docs/xin/Tutorials/Intra-Process-Communication.html)
```

Comment:
This should point to the official docs at https://docs.ros.org/en/rolling/Tutorials/Demos/Intra-Process-Communication.html